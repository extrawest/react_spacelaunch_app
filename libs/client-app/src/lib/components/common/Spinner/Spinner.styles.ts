import type { SxProps, Theme } from '@mui/material';
import { alpha } from '@mui/material';

const wrapper: SxProps<Theme> = (theme) => ({
  position: 'fixed',
  inset: 0,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  zIndex: 'modal',
  bgcolor: alpha(theme.palette.background.default, 0.7),
});

export const styles = {
  wrapper,
};
