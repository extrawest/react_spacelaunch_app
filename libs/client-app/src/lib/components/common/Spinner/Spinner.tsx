import { Box, CircularProgress } from '@mui/material';

import { styles } from './Spinner.styles';

export const Spinner = () => {
  return (
    <Box sx={styles.wrapper}>
      <CircularProgress
        color="primary"
        size={75}
        thickness={4.5}
      />
    </Box>
  );
};
