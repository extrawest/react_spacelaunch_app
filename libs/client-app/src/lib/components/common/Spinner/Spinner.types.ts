export interface SpinnerProps {
  variant?: 'fixed' | 'relative';
}

export interface WrapperStylesOptions {
  fixed: boolean;
}
