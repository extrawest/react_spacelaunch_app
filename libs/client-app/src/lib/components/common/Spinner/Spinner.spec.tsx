import { render } from '@testing-library/react';

import { Spinner } from './Spinner';

describe('[component]: Spinner', () => {
  test('Should match default variant', () => {
    const { baseElement } = render(<Spinner />);

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  test('Should match fixed variant', () => {
    const { baseElement } = render(<Spinner variant="fixed" />);

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
