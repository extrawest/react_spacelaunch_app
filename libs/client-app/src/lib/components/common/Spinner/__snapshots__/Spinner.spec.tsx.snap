// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component]: Spinner Should match default variant 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-86397r"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 75px; height: 75px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.75"
            stroke-width="4.5"
          />
        </svg>
      </span>
    </div>
  </div>
</body>
`;

exports[`[component]: Spinner Should match fixed variant 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-11kauw"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 75px; height: 75px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.75"
            stroke-width="4.5"
          />
        </svg>
      </span>
    </div>
  </div>
</body>
`;
